cmip6_downscaling/workflows/gard_flow.py

```diff
@@ -0,0 +1,209 @@
+import os
+
+os.environ['PREFECT__FLOWS__CHECKPOINTING'] = 'true'
+
+from typing import List
+
+import xarray as xr
+from prefect import Flow, Parameter, task
+from xpersist.prefect.result import XpersistResult
+
+from cmip6_downscaling.config.config import (
+    intermediate_cache_store,
+    results_cache_store,
+    serializer,
+)
+from cmip6_downscaling.data.observations import get_obs
+from cmip6_downscaling.methods.gard import gard_fit_and_predict, gard_postprocess, generate_scrf
+from cmip6_downscaling.tasks.common_tasks import (
+    bias_correct_gcm_task,
+    bias_correct_obs_task,
+    coarsen_and_interpolate_obs_task,
+    interpolate_gcm_task,
+    path_builder_task,
+)
+from cmip6_downscaling.workflows.paths import (
+    make_bias_corrected_gcm_path,
+    make_gard_post_processed_output_path,
+    make_gard_predict_output_path,
+    make_rechunked_obs_path,
+    make_scrf_path,
+)
+from cmip6_downscaling.workflows.utils import rechunk_zarr_array_with_caching
+
+fit_and_predict_task = task(
+    gard_fit_and_predict,
+    checkpoint=True,
+    result=XpersistResult(intermediate_cache_store, serializer=serializer),
+    target=make_gard_predict_output_path,
+)
+
+
+generate_scrf_task = task(
+    generate_scrf,
+    result=XpersistResult(intermediate_cache_store, serializer=serializer),
+    target=make_scrf_path,
+)
+
+
+gard_postprocess_task = task(
+    gard_postprocess,
+    result=XpersistResult(results_cache_store, serializer=serializer),
+    target=make_gard_post_processed_output_path,
+)
+
+
+@task(nout=3)
```

Comment:
AFAIK, if you want xpersist caching, each task can only have a single return. Could this be split into three 'prep input' tasks for caching?

Reply:
ah yup, this task is not cached separately because the rechunkers have built in caching already. we can split it later if needed but i think it's fine to leave this for now